mitodl/mit-open
# content file search

# What are the relevant tickets?
closes https://github.com/mitodl/mit-open/issues/142

# Description (What does it do?)
This pr loads content file data into the open search index and creates a new import for content file search at 
http://localhost:8063/api/v1/content_file_search

The available parameters are
- aggregations array[string] | Available values : topic, file_content_type
- file_content_type: Currently this is always null in the index because only ocw files have this field. In the future it will contain the file content type such as "Assignment" or "Lecture notes"
- id array[string]
- limit integer
- offset integer
- q string - the query string
- sortby string | Available values : id, -id, resource_readable_id, -resource_readable_id
-  topic array[string]

Since we can now have data from the same source with different platforms, I also updated the update_index command to have an etl_source flag instead of a platform flag.

# How can this be tested?
- Recreate the index `docker-compose run web ./manage.py recreate_index --all`
- Verify that management commands that import file data still work, for example `docker-compose run web ./manage.py
backpopulate_xpro_files`
- Verify that the update index command works `docker-compose run web ./manage.py update_index --all`. Also try using the course_etl_source flag. For example `docker-compose run web ./manage.py update_index --content_files --course_etl_source xpro`
- Go to  http://localhost:8063/api/v1/content_file_search/ and play around with the search and check that you get expected results
- Go to  http://localhost:8063/api/v1/learning_resources_search/ and play around with the search and check that you get expected results
